### PR TITLE
Feature/user local time

### PR DESCRIPTION
**Pull request by s3341458**

### CODE REVIEW

File: typescript/src/entities/user.ts [hunk 1]
```diff
@@ -95,120 +96,128 @@ export class User extends Entity {
...
-  @User.property({arrayType: "Invoice"})
+  @User.property({ arrayType: "Invoice" })
   public invoicesHas?: Array<Invoice>;
 
-  @User.property({arrayType: "DomainInvitation"})
+  @User.property({ arrayType: "DomainInvitation" })
   public sentDomainInvitations?: Array<DomainInvitation>;
 
-  @User.property({arrayType: "DomainInvitation"})
+  @User.property({ arrayType: "DomainInvitation" })
```
Comment (DanielCollins) [inline typescript/src/entities/user.ts]: this has nothing to do with times formatting.

File: typescript/src/entities/user.ts [hunk 2]
```diff
@@ -95,120 +96,128 @@ export class User extends Entity {
...
-  @User.property({arrayType: "Theme"})
+  @User.property({ arrayType: "Theme" })
   public themes?: Array<Theme>;
 
-  @User.property({arrayType: "ProductionComment"})
+  @User.property({ arrayType: "ProductionComment" })
   public forwardedProductionComments?: Array<ProductionComment>;
+
+  public userLocalTimeFormat = (time: Date, formatString?: string): string => {
+    var format = formatString ? formatString : "ddd Do MMM YY";
```
Comment (DanielCollins) [inline typescript/src/entities/user.ts]: formatting times for display is a display issue. it probably belongs in a view layer and not an api wrapper layer

File: typescript/src/entities/user.ts (hunk 1 above)
Comment (s3341458) [reply typescript/src/entities/user.ts]: I am so sorry about this as this is propably comes from my auto syntax fixer plugin "ale > prettier", I was totally unaware of this as I set it fix things for me aggressively.

I am willing to a make another pr for syntax issue

however as this pr has already been polluted, I am afraid I do not spend time manually rollback what prettier has already done for time saving reason.

File: typescript/src/entities/user.ts (hunk 2 above)
Comment (s3341458) [reply typescript/src/entities/user.ts]: > formatting times for display is a display issue. it probably belongs in a view layer and not an api wrapper layer

this is very true, however whether sdk is only a api wrapper is questionable. If we do not provide this thing in sdk, someone has implemented it in view level, which to me is very human, as it is our decision to make our time data timezone aware.  Sdk user should know as less as possible, also there should be function lays in sdk to indicate how to proper formating time.


In short, telling sdk user "we give you a UTC time and give you timezone feel free to format the time by yourself, sorry do you know how to convert? sorry you do not deserve to use this sdk" is not nature to me.

**Comment by s3341458** [comment]
credit to @DanielCollins, after think it twice, I think @DanielCollins's suggestion is somehow very reasonable.
